### DISCUSSION TITLE
have background service or other windows with own code base

### DISCUSSION THREAD

**Discussion by Dexus** [alex8088/electron-vite#686] [Q&A]
Hello, I have an app that currently uses multiple windows, now I would like to use a separate renderer for each window, but I currently do not know how I can do that. At least I get an error when I try to add another window/renderer, because of this type:

```

interface ElectronViteConfig {
    /**
     * Vite config options for electron main process
     *
     * https://vitejs.dev/config/
     */
    main?: UserConfigExport;
    /**
     * Vite config options for electron renderer process
     *
     * https://vitejs.dev/config/
     */
    renderer?: UserConfigExport;
    /**
     * Vite config options for electron preload files
     *
     * https://vitejs.dev/config/
     */
    preload?: UserConfigExport;
}

```

Is there a way to archive this?

**Comment by litingyes** [comment]
https://github.com/alex8088/electron-vite/discussions/522#discussioncomment-9642890
